chains/ethereum/connection.go

```diff
@@ -56,7 +56,13 @@ func NewConnection(cfg *Config, kp *secp256k1.Keypair) *Connection {
 // Connect starts the ethereum WS connection
 func (c *Connection) Connect() error {
 	log15.Info("Connecting to ethereum chain...", "chain", c.cfg.name, "url", c.cfg.endpoint)
-	rpcClient, err := rpc.DialWebsocket(c.ctx, c.cfg.endpoint, "/ws")
+	var rpcClient *rpc.Client
+	var err error
+	if c.cfg.http {
+		rpcClient, err = rpc.DialHTTP(c.cfg.endpoint)
+	} else {
+		rpcClient, err = rpc.DialWebsocket(c.ctx, c.cfg.endpoint, "/ws")
+	}
```

Comment:
I envision this could be a bit more robust.  Try connecting via web-sockets first, if that has an error, attempt connecting to http. If that fails then fully error out. 

cc: @ansermino thoughts?

Reply:
That doesn't really add robustness. I think it would be better to have control over which you want to be using.

Reply:
@GregTheGreek

Reply:
yeah i guess that makes sense as well